deepchecks/checks/performance/model_error_analysis.py

```diff
@@ -0,0 +1,161 @@
+            y_pred = model.predict(dataset.features_columns)
+            score = list(map(lambda x, y: mean_squared_error([x], [y]), dataset.label_col, y_pred))
+
+        numeric_transformer = SimpleImputer()
+        categorical_transformer = Pipeline(
+            steps=[('imputer', SimpleImputer(strategy='most_frequent')), ('encoder', TargetEncoder(cat_features))]
+        )
+
+        numeric_features = [num_feature for num_feature in dataset.features if num_feature not in dataset.cat_features]
+
+        preprocessor = ColumnTransformer(
+            transformers=[
+                ('num', numeric_transformer, numeric_features),
+                ('cat', categorical_transformer, cat_features),
+            ]
+        )
+
+        error_model = Pipeline(steps=[
+            ('preprocessing', preprocessor),
+            ('model', RandomForestRegressor(max_depth=4, n_jobs=-1, random_state=self.random_seed))
+        ])
+
+        error_model.fit(dataset.features_columns, y=score)
+
+        error_fi = calculate_feature_importance(error_model, dataset)
+        error_fi.sort_values(ascending=False, inplace=True)
+
+        display_data = dataset.data.assign(score=score)
+
+        min_score = min(score)
+        max_score = max(score)
+
+        def display_categorical(data, feature_name, groupby):
+            plt.figure(figsize=(10, 7))
+            ax = plt.gca()
+            categories = []
+            all_values = []
+            for category, rows in groupby.groups.items():
+                cat_data = data.iloc[rows]
+                all_values.append(cat_data['score'])
+                categories.append(category)
+
+            ax.violinplot(all_values, showextrema=False,showmedians=True)
+            ax.xaxis.set_major_locator(mticker.FixedLocator(range(0, len(categories) + 1)))
+            ax.set_xticklabels([''] + categories)
+
+            plt.xticks(rotation=30)
+            plt.title(feature_name)
+
+        def display_numeric(data, feature_name):
+            plt.figure(figsize=(10, 7))
+            cm = plt.cm.get_cmap('RdYlBu_r')
+
+            sc = plt.scatter(x=feature_name, y='score', data=data, alpha=0.1, edgecolors='none',
+                             c=data['score'], vmin=min_score, vmax=max_score, cmap=cm)
+            plt.colorbar(sc)
+            plt.title(feature_name)
+
+        display = []
+
+        for feature in error_fi.keys()[:self.max_features]:
+            if error_fi[feature] < self.min_error:
+                break
+
+            if feature in cat_features:
+                feat = dataset.data[feature]
+                scored_feature = pd.DataFrame(feat).assign(score=score)
+                grouped_category = scored_feature.groupby(feat)
+                display.append(partial(display_categorical, display_data, feature, grouped_category))
+            else:
+                display.append(partial(display_numeric, display_data, feature))
+
+        value = None
+        headnote = """<span>
```

Comment:
This explanation should talk way more on what we're actually doing here, it's non trivial